sentry-android-core/src/main/java/io/sentry/android/core/AndroidUserCache.java

```diff
@@ -0,0 +1,90 @@
+package io.sentry.android.core;
+
+import android.content.SharedPreferences;
+import io.sentry.core.IUserCache;
+import io.sentry.core.protocol.User;
+import io.sentry.core.util.Objects;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+// TODO: to think, this could be a SentryOptionsCache that could be expanded, does it make sense?
+
+/** the Android User Cache class. It caches the user data in the Android SharedPreferences */
+public final class AndroidUserCache implements IUserCache {
+
+  // TODO: only these 3 fields make sense to me, otherwise caching a raw json would be better, but I
+  // don't want to do that.
+  private static final String EMAIL = "email";
```

Comment:
Shouldn't we prefix those constants here? Could it be that the users of our SDK also save email to their shared preferences? What about 'sentry_user_cache_email' or something similar?

Reply:
its a dedicated SharedPreferences, see:
https://github.com/getsentry/sentry-android/pull/342/files#diff-5b8dbf3e5fc61d592bc69f90d2a3b53cR76-R77